### PR TITLE
Add QCObjects via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by jeanmachuca**
Adding QCObjects to cdnjs

# Pull request template for new library

Pull request for issue: #13679 
Related issue(s): #13679 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required):
* Official website (optional, not the repository):
* NPM package url (optional):
* License and its reference:
* GitHub / Bitbucket popularity (required):
  * Count of watchers:
  * Count of stars:
  * Count of forks:
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week:
  * Downloads in the last month:

## Essential checklist

* [x] I'm the author of this library
  * [x] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/qcobjects/package.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "qcobjects",
+  "version": "2.1.298",
+  "description": "QCObjects a Quickcorp Objects Library - Cross Browser Javascript Framework for MVC Patterns",
+  "main": "qcobjects.min.js",
+  "license": "LGPL-3.0",
+  "scripts": {
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]: Attr not needed

File: ajax/libs/qcobjects/package.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    "n-tier",
+    "multitier",
+    "multi-tier"
+  ],
+  "author": "Jean Machuca <correojean@gmail.com>",
+  "bugs": {
+    "url": "https://github.com/QuickGroup/QCObjects/issues"
+  },
+  "homepage": "https://github.com/QuickGroup/QCObjects#readme",
+  "dependencies": {
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]: Attr not needed

File: ajax/libs/qcobjects/package.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    "qcobjects-cli": "latest",
+    "qcobjects-sdk": "latest"
+  },
+  "devDependencies": {
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]: Attr not needed

File: ajax/libs/qcobjects/package.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    "jasmine": "^3.4.0"
+  },
+  "autoupdate": {
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]: You only need one auto-update config, either git or npm.

File: ajax/libs/qcobjects/package.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "qcobjects",
+  "version": "2.1.298",
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]: Version should be omitted unless you wish to include the correct files for this version in the PR.

File: ajax/libs/qcobjects/package.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  "description": "QCObjects a Quickcorp Objects Library - Cross Browser Javascript Framework for MVC Patterns",
+  "main": "qcobjects.min.js",
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]:
```suggestion
  "filename": "qcobjects.min.js",
```
This attr should be named `filename`, not `main`.

Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]:
```suggestion
  "main": "QCObjects.js",
```
This file needs to exist in the npm package, it appears only the unminfiied version `QCObjects.js` does.

File: ajax/libs/qcobjects/package.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  "license": "LGPL-3.0",
+  "npmName": "qcobjects",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "./QCObjects.js"
```
Comment (MattIPv4) [inline ajax/libs/qcobjects/package.json]:
```suggestion
        "QCObjects.js"
```
Glob doesn't require './'

File: ajax/libs/qcobjects/package.json (hunk 5 above)
Comment (jeanmachuca) [reply ajax/libs/qcobjects/package.json]: Cool, I've added homepage, repository, bugs, author and keywords attributes

File: ajax/libs/qcobjects/package.json (hunk 6 above)
Comment (jeanmachuca) [reply ajax/libs/qcobjects/package.json]: Done.

Comment (jeanmachuca) [reply ajax/libs/qcobjects/package.json]: QCObjects doesn't need to be minified as the dynamic packages loading model is meant to warranty split and one-demand loading of scripts resources. If it is an explicit requirement for cdnjs I could minify the main file QCObjects.js for compatibility purposes. In the meantime, I've just changed the name to point to the right one (QCObjects.js). Cheers.

File: ajax/libs/qcobjects/package.json (hunk 7 above)
Comment (jeanmachuca) [reply ajax/libs/qcobjects/package.json]: Done

File: ajax/libs/qcobjects/package.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	},
+	"homepage": "https://github.com/QuickGroup/QCObjects#readme",
+	"npmName": "qcobjects",
+	"npmFileMap": [{
+		"basePath": "",
+		"files": [
+			"QCObjects.js"
+		]
+	}]
+}
```
Comment (xtuc) [inline ajax/libs/qcobjects/package.json]: Looks like the formatting is inconcistent with the other package.json

Comment (MattIPv4) [reply ajax/libs/qcobjects/package.json]: Looks to be valid, just that `npmFileMap` has the array and object openings on the same line?

Comment (xtuc) [reply ajax/libs/qcobjects/package.json]: Are we not using 2 spaces instead of 4?

Comment (MattIPv4) [reply ajax/libs/qcobjects/package.json]: Oh yeah, this is indented with tabs it looks like, should be 2 spaces, you're right.

Comment (jeanmachuca) [reply ajax/libs/qcobjects/package.json]: Ok, I've changed the indent to 2 spaces

**Comment by MattIPv4** [comment]
Don't we want keywords? They help Algolia with search matches and provide meta-data for library pages?

**Comment by xtuc** [comment]
ah right good catch. I wish we had the bot to double check that.

**Comment by jeanmachuca** [comment]
> ah right good catch. I wish we had the bot to double check that.

So I guess isn't anything else to do?

**Pull request merged**